ipaserver/plugins/certmap.py

```diff
@@ -0,0 +1,345 @@
+
+
+@register()
+class certmaprule(LDAPObject):
+    """
+    Certificate Identity Mapping Rules
+    """
+
+    label = _('Certificate Identity Mapping Rules')
+    label_singular = _('Certificate Identity Mapping Rule')
+
+    object_name = _('Certificate Identity Mapping Rule')
+    object_name_plural = _('Certificate Identity Mapping Rules')
+    object_class = ['ipacertmaprule']
+
+    container_dn = api.env.container_certmaprules
+    default_attributes = [
+        'cn', 'description', 'ipacertmapissuer',
+        'ipacertmapmaprule',
+        'ipacertmapmatchrule',
+        'associateddomain',
+        'ipacertmappriority',
+        'ipaenabledflag'
+    ]
+    search_attributes = [
+        'cn', 'description', 'ipacertmapissuer',
+        'ipacertmapmaprule',
+        'ipacertmapmatchrule',
+        'associateddomain',
+        'ipacertmappriority',
+        'ipaenabledflag'
+    ]
+
+
+    takes_params = (
+        Str('cn',
+            cli_name='rulename',
+            primary_key=True,
+            label=_('Rule name'),
+            doc=_('Certificate Identity Mapping Rule name'),
+        ),
+        Str('description?',
+            cli_name='desc',
+            label=_('Description'),
+            doc=_('Certificate Identity Mapping Rule description'),
+        ),
+        DNParam('ipacertmapissuer?',
+            cli_name='issuer',
+            label=_('Issuer'),
+            doc=_('LDAP DN of the certificate issuer (CN=Certificate Authority,O=DOMAIN.COM)'),
+        ),
+        Str('ipacertmapmaprule?',
+            cli_name='maprule',
+            label=_('Mapping rule'),
+            doc=_('Rule used to map the certificate with a user entry'),
+        ),
+        Str('ipacertmapmatchrule?',
+            cli_name='matchrule',
+            label=_('Matching rule'),
+            doc=_('Rule used to check if a certificate can be used for authentication'),
+        ),
+        Str('associateddomain*',
+            _domain_name_validator,
+            normalizer=_domain_name_normalizer,
+            cli_name='domain',
+            label=_('Domain name'),
+            doc=_('Domain where the user entry will be searched'),
+        ),
+        Int('ipacertmappriority?',
+            cli_name='priority',
+            label=_('Priority'),
+            doc=_('Priority of the rule (higher number means lower priority'),
+            minvalue=0,
+        ),
+        Flag('ipaenabledflag?',
+            label=_('Enabled'),
+            flags=['no_option'],
+            default=True
+        ),
+    )
+
+    permission_filter_objectclasses = ['ipacertmaprule']
+    managed_permissions = {
+        'System: Add Certmap Rules': {
+            'replaces_global_anonymous_aci': True,
+            'ipapermright': {'add'},
+            'default_privileges': {
+                'Certificate Identity Mapping Administrators'},
+        },
+        'System: Read Certmap Rules': {
```

Comment:
I have a question related to ACI, if `System: Read Certmap Configuration` is allowed for `ldap://all`, shouldn't be this `System: Read Certmap Rules` accessible to `ldap://all`? What/who is the consumer of this rules? SSSD?